src/assertions/iterator.rs

```diff
@@ -621,12 +588,52 @@ where
     }
 }
 
+pub(crate) fn add_input_iter_fact<A: Debug, IA: Iterator<Item = A> + Clone>(
+    fact_name: &str,
+    result: &mut AssertionResult,
+    iter: IA,
+) -> AssertionResult {
+    let use_multiline_output = iter
+        .clone()
+        .map(|x| format!("{:?}", x).len())
+        .max_by(|x, y| x.cmp(y))
+        .unwrap_or(0)
+        > DEBUG_LENGTH_WRAP_LIMIT;
+    let formatted = format!(
+        "{}",
+        if use_multiline_output {
+            let elements = iter
+                .clone()
+                .map(|el| format!("  - {:?}", el))
```

Comment:
Hmm.  Formatting logic here is not good as every assertion implementations need to own similar formatting logic.

I believe there is no reason to keep using [key-value fact](https://github.com/google/assertor/blob/6addc641a5c756e973f2863e6222054c72767da6/src/base.rs#L417). 
 This is good timing to introduce a new fact for keyed-values; for example, `KeyValue**s** { key: String, values: Vec<String>}`.  Making multiline can be done when stringifying the facts [here](https://github.com/google/assertor/blob/6addc641a5c756e973f2863e6222054c72767da6/src/base.rs#L345-L355).  This makes easier to write tests and to support a command line option to disable multiline if needed.

This implementation is re-usable when supporting Maps depending on its desired behavior (let's discuss in #30).

Let me know if you have any other ideas. :)

Reply:
I was a bit hesitant to change the core facts enum so I appreciate your guidance 👍  Lmk how the new implementation looks.